code/core/src/androidTest/java/com/adobe/marketing/mobile/services/ui/message/views/MessageScreenTests.kt

```diff
@@ -316,6 +336,688 @@ class MessageScreenTests {
+        rootBounds.left.assertIsEqualTo(frameBounds.left, "failed", Dp(2f))
+        rootBounds.right.assertIsEqualTo(frameBounds.right, "failed", Dp(2f))
+        assertTrue(rootBounds.left < contentBounds.left)
+        rootBounds.right.assertIsEqualTo(contentBounds.right, "failed", Dp(2f))
+    }
+
+    // ----------------------------------------------------------------------------------------------
+    // Test cases for alignment with insets
+    // ----------------------------------------------------------------------------------------------
+    @Test
+    fun testMessageScreenAlignmentWhenTopAlignedWithPositiveInset() {
+        val offsetPercent = 20
+        val heightPercent = 60
+        val settings = InAppMessageSettings.Builder()
+            .content(HTML_TEXT_SAMPLE)
+            .height(heightPercent)
+            .width(80)
+            .verticalAlignment(InAppMessageSettings.MessageAlignment.TOP)
+            .verticalInset(offsetPercent)
+            .build()
+
+        var screenHeightDp: Dp = 0.dp
+        var screenWidthDp: Dp = 0.dp
+        val activityHeightDp = getActivityHeightInDp(composeTestRule.activity)
+        composeTestRule.setContent { // setting our composable as content for test
+            val currentConfiguration = LocalConfiguration.current
+            screenHeightDp = currentConfiguration.screenHeightDp.dp
+            MessageScreen(
+                presentationStateManager = presentationStateManager,
+                inAppMessageSettings = settings,
+                onCreated = { onCreatedCalled = true },
+                onDisposed = { onDisposedCalled = true },
+                onGestureDetected = { gesture -> detectedGestures.add(gesture) },
+                onBackPressed = { onBackPressed = true }
+            )
+        }
+
+        composeTestRule.waitForIdle()
+        composeTestRule.onNodeWithTag(MessageTestTags.MESSAGE_FRAME).assertDoesNotExist()
+
+        // Change the state of the presentation state manager to shown to display the message
+        presentationStateManager.onShown()
+        composeTestRule.waitForIdle()
+        validateMessageExistsWhenClipped(false)
+
+        val frameBounds = composeTestRule.onNodeWithTag(MessageTestTags.MESSAGE_FRAME).getUnclippedBoundsInRoot()
+        val contentBounds = composeTestRule.onNodeWithTag(MessageTestTags.MESSAGE_CONTENT).getUnclippedBoundsInRoot()
+        val offsetDp = screenHeightDp * offsetPercent.toFloat() / 100f
+        val contentHeightDp = screenHeightDp * heightPercent.toFloat() / 100f
+
+        // Frame and content is offset downwards from the top edge of the activity by the inset value
+        frameBounds.top.assertIsEqualTo(offsetDp, "failed", Dp(2f))
+        frameBounds.bottom.assertIsEqualTo(activityHeightDp + offsetDp, "failed", Dp(2f))
+        contentBounds.top.assertIsEqualTo(offsetDp, "failed", Dp(2f))
+        contentBounds.bottom.assertIsEqualTo(contentHeightDp + offsetDp, "failed", Dp(2f))
```

Comment:
Can we move these calculations to a separate method to be used by all alignment tests?